File: python/algorithms/ShellSort.py
def shellSort(input_list):
    gap = len(input_list) / 2
    while gap > 0:
        for i in range(gap, len(input_list)):
            temp = input_list[i]
            j = i
        
# Sort sub list for this gap
            while j >= gap and input_list[j - gap] > temp:
                input_list[j] = input_list[j - gap]
                j = j-gap
            input_list[j] = temp

# Reduce the gap for the next element
        gap = gap/2

list = [5,2,5531,45,66,11,121,99]

shellSort(list)
print(list)

File: javascript/Sorting/shellSort.js
const arrayToSort = [234,544,235,1,4,555,4435,1223,545];
const gaps = [701, 301, 132, 57, 23, 10, 4, 1];
// Mutilates the array so be careful
const shellSort = (array) => {
  for(var g = 0; g < gaps.length; g++) {
    let gap = gaps[g];
    for(var i = gap; i < array.length; i++) {
      var temp = array[i];
      for(var j = i; j >= gap && array[j - gap] > temp; j -= gap) {
        array[j] = array[j - gap];
      }
      array[j] = temp;
    }
  }
  return array;
}

console.log(`Your initial array is: ${arrayToSort}`);
console.log(`After shell sort: ${shellSort(arrayToSort)}`);